### PR TITLE
Misc. Power linkage cleanup

### PR DESCRIPTION
**Pull request by 0xdaryl**
* Remove and fold calculateParameterRegisterOffset from Linkage
* Remove and fold calculateActualParameterOffset from Linkage
* Fold and simplify code in Power linkage saveArguments
* Unify float/double handling in Power linkage

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@genie-omr build plinux

This is a Power-specific change.  A full genie build is not required.

**Comment by 0xdaryl** [comment]
@ymanton @gita-omr, I'd appreciate a review from at least one of you please.

File: compiler/p/codegen/OMRLinkage.cpp [hunk 1]
```diff
@@ -136,35 +133,33 @@ TR::Instruction *OMR::Power::Linkage::saveArguments(TR::Instruction *cursor, boo
...
             {
             regNum = properties.getFloatArgumentRegister(lri);
-            if (paramCursor->isReferencedParameter()) freeScratchable.reset(regNum);
-            if (twoRegs)
-               if (paramCursor->isReferencedParameter()) freeScratchable.reset(regNum+1);
             }
+
+         if (paramCursor->isReferencedParameter()) freeScratchable.reset(regNum);
+         if (twoRegs)
+            if (paramCursor->isReferencedParameter()) freeScratchable.reset(regNum+1);
```
Comment (ymanton) [inline compiler/p/codegen/OMRLinkage.cpp]: Maybe this instead?

```
if (paramCursor->isReferencedParameter())
   {
   freeScratchable.reset(regNum);
   if (twoRegs)
      freeScratchable.reset(regNum+1);
   }
```

Comment (0xdaryl) [reply compiler/p/codegen/OMRLinkage.cpp]: Agreed.  Fixed.

**Comment by 0xdaryl** [comment]
@genie-omr build plinux

**Comment by 0xdaryl** [comment]
Reviews and testing have completed.  Would another committer (e.g., @Leonardo2718 @fjeremic ) mind merging this please?

**Pull request merged**